### PR TITLE
system variable: add tidb_allow_fallback_to_tikv

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add `tidb_allow_fallback_to_tikv` into system variables.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): pingcap/tidb#22459, pingcap/tidb#22886

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@xuyifangreeneyes Could you specify from which version this variable is introduced?

**Comment by xuyifangreeneyes** [comment]
> @xuyifangreeneyes Could you specify from which version this variable is introduced?

v5.0

**Comment by TomShawn** [comment]
> > @xuyifangreeneyes Could you specify from which version this variable is introduced?
> 
> v5.0

Do you mean v5.0 GA?

**Comment by xuyifangreeneyes** [comment]
> > > @xuyifangreeneyes Could you specify from which version this variable is introduced?
> > 
> > 
> > v5.0
> 
> v5.0 GA or v5.0 RC?

v5.0 GA

File: system-variables.md [hunk 1]
```diff
@@ -1101,3 +1101,9 @@ set tidb_slow_log_threshold = 200;
 - 作用域：SESSION | GLOBAL
 - 默认值：ON
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
+
+### `tidb_allow_fallback_to_tikv`
```
Comment (TomShawn) [inline system-variables.md:1105]:
```suggestion
### `tidb_allow_fallback_to_tikv`  <span class="version-mark">从 v5.0 GA 版本开始引入</span>

```

**Review by winoros — COMMENTED** [review]
lgtm

**Review by qw4990 — APPROVED** [review]

**Review by qw4990 — COMMENTED** [review]
/LGTM

File: system-variables.md [hunk 2]
```diff
@@ -1101,3 +1101,9 @@ set tidb_slow_log_threshold = 200;
 - 作用域：SESSION | GLOBAL
 - 默认值：ON
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
+
+### `tidb_allow_fallback_to_tikv`  <span class="version-mark">从 v5.0 GA 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 默认值：""
+- 当 SQL 语句因为该变量指定的存储引擎的故障而执行失败时，TiDB 会尝试使用 TiKV 再次执行该 SQL 语句。目前支持设置该变量为 "" 或者 "tiflash"。当该变量被设置为 "tiflash" 时，TiDB 会在 TiFlash 发生故障导致 SQL 语句执行失败的情况下尝试使用 TiKV 再次执行。
```
Comment (qiancai) [inline system-variables.md:1109]:
```suggestion
- 这个变量用于指定一个存储引擎。当该变量指定的存储引擎发生故障导致 SQL 语句执行失败时，TiDB 会尝试使用 TiKV 再次执行该 SQL 语句。目前支持设置该变量为 "" 或者 "tiflash"。如果设置该变量为 "tiflash" ，当TiFlash 发生故障导致 SQL 语句执行失败时，TiDB 会尝试使用 TiKV 再次执行该 SQL 语句。
```

**Comment by qiancai** [comment]
@xuyifangreeneyes, I added a suggestion on the doc changes. PTAL, thanks~

File: system-variables.md [hunk 3]
```diff
@@ -1119,3 +1119,9 @@ set tidb_slow_log_threshold = 200;
 - 作用域：SESSION | GLOBAL
 - 默认值：ON
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
+
+### `tidb_allow_fallback_to_tikv`  <span class="version-mark">从 v5.0 GA 版本开始引入</span>
```
Comment (TomShawn) [inline system-variables.md:1123]:
```suggestion
### `tidb_allow_fallback_to_tikv` <span class="version-mark">从 v5.0 GA 版本开始引入</span>
```

File: system-variables.md [hunk 4]
```diff
@@ -1119,3 +1119,9 @@ set tidb_slow_log_threshold = 200;
...
+
+- 作用域：SESSION | GLOBAL
+- 默认值：""
+- 这个变量用于指定一个存储引擎。当该变量指定的存储引擎发生故障导致 SQL 语句执行失败时，TiDB 会尝试使用 TiKV 再次执行该 SQL 语句。目前支持设置该变量为 "" 或者 "tiflash"。如果设置该变量为 "tiflash" ，当TiFlash 发生故障导致 SQL 语句执行失败时，TiDB 会尝试使用 TiKV 再次执行该 SQL 语句。
```
Comment (TomShawn) [inline system-variables.md:1127]:
```suggestion
- 这个变量用于指定一个存储引擎。当该变量指定的存储引擎发生故障导致 SQL 语句执行失败时，TiDB 会尝试使用 TiKV 再次执行该 SQL 语句。目前支持设置该变量为 "" 或者 "tiflash"。如果设置该变量为 "tiflash"，当 TiFlash 发生故障导致 SQL 语句执行失败时，TiDB 会尝试使用 TiKV 再次执行该 SQL 语句。
```

Comment (xuyifangreeneyes) [reply system-variables.md:1127]: - 执行的主体是 TiDB 。感觉更准确的说法是 TiDB 会尝试将 TiKV 作为存储引擎执行 SQL 语句。
- 该变量于 tidb_isolation_read_engines 类似，可以指定若干存储引擎（目前的表述不太相符，我更改一下），比如未来可能有某个新的存储引擎，假设是 mem ，则 tidb_allow_fallback_to_tikv 可以设置为 "tiflash, mem"。"" 表示没有任何存储引擎开启 fallback 的行为。
- 更准确的说法是这个变量用于指定若干存储引擎。当任一指定的存储引擎不可用时，TiDB 会尝试将 TiKV 作为存储引擎再次执行 SQL 语句。

**Comment by xuyifangreeneyes** [comment]
@TomShawn PTAL

File: system-variables.md (hunk 4 above)
Comment (TomShawn) [reply system-variables.md]: Please align the change to https://github.com/pingcap/docs/pull/4781.

**Pull request merged**